cypress/integration/lsp-tokens/3-settle.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable */
+import {
+  LSP_PAIRNAME,
+  TEST_PUBLIC_ADDRESS,
+  COLLATERAL_TO_MINT,
+  COLLATERAL_EXPECTED_AFTER_MINT,
+  COLLATERAL_TO_REDEEM,
+  COLLATERAL_EXPECTED_AFTER_REDEEM,
+} from "../../utils/constants";
+import deployLSPContract from "../../utils/deployLSPContract";
+
+/* 
+  Note: To run this test you need a few things running:
+  1) Dev API
+  2) Hardhat node, forking mainnet
+  3) Main app
+  4) You need to make sure the test account has UMA: see uma-protocol/hardhat-test and run the seedUmaToAccounts.
+  More details in the README.
+  5) Hardhat needs to be moved forward past the settle date.
+*/
+
+describe("Connects to the wallet", () => {
+  let lspAddress = "0xfd7Ead07dF3cD2543fE269d9E320376c64D9143E";
+  before(async () => {
+    lspAddress = await deployLSPContract();
+  });
+
+  beforeEach(() => {
+    // Cypress starts out with a blank slate for each test
+    // so we must tell it to visit our website with the `cy.visit()` command.
+    // Since we want to visit the same URL at the start of all our tests,
+    // we include it in our beforeEach function so that it runs before each test
+  });
+
+  // it("Move blockchain forward to a settable state", () => {
+  //   cy.exec(
+  //     "HARDHAT_NETWORK=localhost node ./hardhat-scripts/advanceTimeForward.js 173581840"
+  //   ).then((res) => {
+  //     // Should be no error.
+  //     expect(res.code).to.eq(0);
+  //   });
+  // });
+
+  it("Visits newly minted test contract", () => {
+    // Give API time to load detect the contract.
+    // TODO: Work to get this time down with BE devs.
+    cy.wait(35_000);
```

Comment:
Have we considered running the API _in process_? Not sure how easy that is, but it could make it easier to detect when the contract is ready.

Another (messier) way is to poll the API in the test to be sure it finds the contract before visiting.

Reply:
yeah we might be able to make this a bit more robust. Right now, there is a potential race condition.

Reply:
Yeah, totally fine to leave it this way for now, but curious to hear @daywiss's thoughts on running the API in process.